identity/state.go

```diff
@@ -0,0 +1,54 @@
+package identity
+
+import (
+	"time"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql/localsql/nipost"
+)
+
+type State any
```

Comment:
Would it make sense to specify an interface here instead of `any`?

The interface for instance could have `APIState() pb.IdentityState` as method. Then the big `switch`/`case` in the GRPC handler wouldn't be necessary 🙂